src/schema/sources.ts

```diff
@@ -818,8 +905,42 @@ export const resolvers: IResolvers<any, Context> = {
 
       return getSourceById(ctx, info, sourceId);
     },
+    removeMember: async (
+      _,
+      { sourceId, memberId }: RemoveMemberArgs,
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      await ensureSourcePermissions(
+        ctx,
+        sourceId,
+        SourcePermissions.RemoveMember,
+      );
+      await ctx.con.transaction(async (manager) => {
+        const repo = manager.getRepository(SourceMember);
+        const [member, loggedUser] = await Promise.all([
+          repo.findOneByOrFail({ sourceId, userId: memberId }),
+          repo.findOneByOrFail({ sourceId, userId: ctx.userId }),
+        ]);
+
+        const memberRank = roleRank[member.role];
+        const loggedUserRank = roleRank[loggedUser.role];
+
+        if (memberRank >= loggedUserRank) {
+          throw new ForbiddenError(
+            `You don't have the required permission for this action!`,
+          );
+        }
```

Comment:
Do we want to leave this as is?
Optional could be using the same rank function

Reply:
Good point. Let me extract this as a generic function.